### ISSUE TITLE
Smaller frontlight dialog

### ISSUE THREAD

**Issue by pblasi** [koreader/koreader#5314]
This big window is filling all the screen to only adjust brightness and temperature.
Please make a simpler, smaller, bottom window which allows seeing the actual book page and text so light can be adjusted more properly. 
Thanks!

**Comment by pazos** [comment]
> This big window is filling all the screen to only adjust brightness and temperature.

And the warm light timer / auto stub, which is nice for some users.

> Please make a simpler, smaller, bottom window which allows seeing the actual book page and text so light can be adjusted more properly.

Err, the frontlight affects all widgets drawn on the screen. Why is important to see the book?
Anyways, most of the time you don't need to open the frontlight widget and can just use gestures to control the frontlight.

**Comment by pblasi** [comment]
Fair enough... I guess I'm used to less intrusive coolreader dialog... The thing is I need to open the widget to adjust light properly because swipe gesture for brightness/warmth is not working properly: #5313

**Comment by pblasi** [comment]
> The ability to get reasonable precision from the intensity bar, the min button,[*] the toggle button, and the -1/+1 are all integral to ease of operation. Ymmv.> 

I would just recommend learning from coolreader's light dialog. Please give it a try and use it for while on your kobo, then be honest on your overall experience. You get all these abilities too. The toggle button, the arrows for fine tuning, and overall the swipe control on the bars for both light and temperature, which makes the real difference... everything in a clean, comfortable design. It works like a charm.

Yes, I know koreader is more on the side "The more the better" than "The simpler the better", so it's my problem if I don't like it ;)

**Comment by pblasi** [comment]
> But I can make it behave like a reasonably decent slider without much effort: #5371 

How can I test it? Thanks!

**Comment by Frenzie** [comment]
If you look at files changed in the PR it shows you the changes:

https://github.com/koreader/koreader/pull/5371/files

Those can be applied manually or using view file → raw you can just replace 'em whole.

https://raw.githubusercontent.com/koreader/koreader/b88063015ef9ee8fabaf7b913f52a4b3610c8607/frontend/ui/widget/button.lua

https://raw.githubusercontent.com/koreader/koreader/b88063015ef9ee8fabaf7b913f52a4b3610c8607/frontend/ui/widget/frontlightwidget.lua

(Alternatively, wait a few days until it shows up in a nightly near you. :-P)

**Comment by pblasi** [comment]
Nice! I would say:
1- Make increments of 1, not 5, to make light changes smoother (analog feeling) when swiping.
2- Apply to "Warmth" as well.
3- Make it work the same way on swipe gestures in gesture manager.
Thanks!

**Comment by Frenzie** [comment]
That would need much more clever refreshing (tied in to scroll rate) and preferably also a slider approach that isn't a quick hack on top of this particular existing mechanism. The experience with a change like this makes for a pretty interesting artifact show:
```
diff --git a/frontend/ui/widget/frontlightwidget.lua b/frontend/ui/widget/frontlightwidget.lua
index 9a00dda2..a68bf20d 100644
--- a/frontend/ui/widget/frontlightwidget.lua
+++ b/frontend/ui/widget/frontlightwidget.lua
@@ -48,7 +48,7 @@ function FrontLightWidget:init()
     self.fl_max = self.powerd.fl_max
     self.fl_cur = self.powerd:frontlightIntensity()
     local steps_fl = self.fl_max - self.fl_min + 1
-    self.one_step = math.ceil(steps_fl / 25)
+    self.one_step = math.ceil(steps_fl / 100)
     self.steps = math.ceil(steps_fl / self.one_step)
     if (self.steps - 1) * self.one_step < self.fl_max - self.fl_min then
         self.steps = self.steps + 1
```

But there's also the ProgressWidget. That one already functions like a proper slider, except for the sliding part. (Easily added, but there's some interaction with MovableContainer going on so only if you disable that.)
